src/cfnlint/graph.py

```diff
@@ -0,0 +1,117 @@
+"""
+Helpers for loading resources, managing specs, constants, etc.
+
+Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+SPDX-License-Identifier: MIT-0
+"""
+import re
+import six
+from networkx import networkx
+
+
+class Graph(object):
+    """Models a template as a directed graph of resources"""
+
+    def __init__(self, cfn):
+        """Builds a graph where resources are nodes and edges are explicit (DependsOn) or implicit (Fn::GetAtt, Fn::Sub, Ref)
+        relationships between resources"""
+
+        # Directed graph that allows self loops and parallel edges
+        self.graph = networkx.MultiDiGraph()
+
+        # add all resources in the template as nodes
+        for resourceId in cfn.template['Resources'].keys():
+            self.graph.add_node(resourceId)
+
+        # add edges for "Ref" tags. { "Ref" : "logicalNameOfResource" }
+        refs_paths = cfn.search_deep_keys('Ref')
+        for ref_path in refs_paths:
+            ref_type, source_id = ref_path[:2]
+            target_id = ref_path[-1]
+            if not ref_type == 'Resources':
+                continue
+
+            if isinstance(target_id, (six.text_type, six.string_types, int)) and (self._is_resource(cfn, target_id)):
+                target_resource_id = target_id
+                self.graph.add_edge(source_id, target_resource_id, label='Ref')
+
+        # add edges for "Fn::GetAtt" tags.
+        # { "Fn::GetAtt" : [ "logicalNameOfResource", "attributeName" ] } or { "!GetAtt" : "logicalNameOfResource.attributeName" }
+        getatt_paths = cfn.search_deep_keys('Fn::GetAtt')
+        for getatt_path in getatt_paths:
+            ref_type, source_id = getatt_path[:2]
+            value = getatt_path[-1]
+            if not ref_type == 'Resources':
+                continue
+
+            if isinstance(value, list) and len(value) == 2 and (self._is_resource(cfn, value[0])):
+                target_resource_id = value[0]
+                self.graph.add_edge(source_id, target_resource_id, label='Fn::GetAtt')
+
+            if isinstance(value, (six.string_types, six.text_type)) and '.' in value:
+                target_resource_id = value.split('.')[0]
+                if self._is_resource(cfn, target_resource_id):
+                    self.graph.add_edge(source_id, target_resource_id, label='!GetAtt')

```

Comment:
```suggestion
                    self.graph.add_edge(source_id, target_resource_id, label='GetAtt')
```

Generalizing this one beyond specific syntax because of the visualizing issues with the other specific syntax